packages/peregrine/lib/talons/CheckoutPage/useCheckoutPage.js

```diff
@@ -1,25 +1,48 @@
-import { useCallback, useState } from 'react';
-import { useMutation } from '@apollo/react-hooks';
+import { useCallback, useEffect } from 'react';
+import { useLazyQuery, useMutation, useQuery } from '@apollo/react-hooks';
 
-import { useAwaitQuery } from '../../hooks/useAwaitQuery';
 import { useAppContext } from '../../context/app';
 import { useUserContext } from '../../context/user';
 import { useCartContext } from '../../context/cart';
 
+export const CHECKOUT_STEP = {
+    SHIPPING_ADDRESS: 1,
+    SHIPPING_METHOD: 2,
+    PAYMENT: 3,
+    REVIEW: 4,
+    RECEIPT: 5
+};
+
 export const useCheckoutPage = props => {
-    const { createCartMutation, getCartDetailsQuery } = props;
+    const {
+        mutations: { createCartMutation },
+        queries: { getCheckoutDetailsQuery, getCheckoutStepQuery }
+    } = props;
 
     const [, { toggleDrawer }] = useAppContext();
     const [{ isSignedIn }] = useUserContext();
-    const [
-        { isEmpty },
-        { createCart, getCartDetails, removeCart }
-    ] = useCartContext();
+    const [{ cartId }, { createCart, removeCart }] = useCartContext();
 
     const [fetchCartId] = useMutation(createCartMutation);
-    const fetchCartDetails = useAwaitQuery(getCartDetailsQuery);
+    const [
+        getCheckoutDetails,
+        { data: checkoutData, loading: checkoutLoading }
+    ] = useLazyQuery(getCheckoutDetailsQuery, {
+        // TODO: Purposely overfetch and hit the network until all components
+        // are correctly updating the cache. Will be fixed by PWA-321.
+        fetchPolicy: 'cache-and-network'
+    });
+
+    const { data: stepData, client } = useQuery(getCheckoutStepQuery);
+    const setCheckoutStep = useCallback(
+        step => {
+            client.writeData({ data: { checkoutStep: step } });
```

Comment:
I discussed the UX with @soumya-ashok and she said that a user, upon a refresh, should return to the step that they were last on. So we do need some sort of persistence layer that "remembers" where the user was. And that can come from the server _or_ it can come from some local state that is simply persisted after a user proceeds through a step.

As for when to clear the data, I added an unmount effect that should have set the value, but I would also have expected the user logging out to clear the cache entirely. I'm surprised that it persisted the checkout step. I'll add the `cartId` change suggestion to the effect and I'll test the user log out scenario myself to see what I need to do. Thanks!

Reply:
Ok, instead of taking care of this here I think we should try to solve the larger problem of our cache not resetting on logout. I opened https://jira.corp.magento.com/browse/PWA-457 for this. 

That said, I don't think your step 3 is actually occurring, or at least I don't see how it did. The checkout page resets the step on unmount so when you left the page to add a new item to your cart the step should have been reset to 1.

Reply:
I am able to reproduce it. Let's deal with this in PWA-457.

Reply:
Your video suggests a bug - I'll fix it in the PR if I can.